src/config/ConfigManager.cpp

```diff
@@ -2115,12 +2142,22 @@ std::optional<std::string> CConfigManager::handleWorkspaceRules(const std::strin
 
     auto                     assignRule = [&](std::string rule) -> std::optional<std::string> {
         size_t delim = std::string::npos;
-        if ((delim = rule.find("gapsin:")) != std::string::npos)
-            wsRule.gapsIn = std::stoi(rule.substr(delim + 7));
-        else if ((delim = rule.find("gapsout:")) != std::string::npos)
-            wsRule.gapsOut = std::stoi(rule.substr(delim + 8));
-        else if ((delim = rule.find("bordersize:")) != std::string::npos)
-            wsRule.borderSize = std::stoi(rule.substr(delim + 11));
+        if ((delim = rule.find("gapsin:")) != std::string::npos) {
+            CVarList varlist = CVarList(rule.substr(delim + 7), 0, ' ');
+            wsRule.gapsIn    = CCssGapData();
+            try {
+                wsRule.gapsIn->parseGapData(varlist);
+            } catch (...) { Debug::log(ERR, "Error parsing workspace rule gaps: {}", rule.substr(delim + 7)); }
+        } else if ((delim = rule.find("gapsout:")) != std::string::npos) {
+            CVarList varlist = CVarList(rule.substr(delim + 8), 0, ' ');
+            wsRule.gapsOut   = CCssGapData();
+            try {
+                wsRule.gapsOut->parseGapData(varlist);
+            } catch (...) { Debug::log(ERR, "Error parsing workspace rule gaps: {}", rule.substr(delim + 8)); }
+        } else if ((delim = rule.find("bordersize:")) != std::string::npos)
+            try {
+                wsRule.borderSize = std::stoi(rule.substr(delim + 11));
+            } catch (...) { Debug::log(ERR, "Error parsing workspace rule bordersize: {}", rule.substr(delim + 11)); }
```

Comment:
throw an error here: `return "error string"`

Reply:
Oh this resolved the banner issue as well, thanks.